source/_posts/2015-04-28-open-sourcing-your-apps.markdown

```diff
@@ -0,0 +1,63 @@
+
+It was 95 days ago, and I was sitting before of my computer in NYC. I loaded my terminal, opened [TapTalk](https://taptalk.me) on my phone, tapped on my collegue [Ash's](http://ashfurrow.com) face and held my finger there. I paused, this moment was a long time, it was worth waiting. I switched the camera from my face to the screen with the terminal open.
+
+This moment was probably the most nervous I had been in years. It showed `git push origin master`. I said "it's happening." I hit return. A new era in the Artsy mobile team had started. A few minutes later, I wrote this tweet:
+
+<center>
+<blockquote class="twitter-tweet" data-cards="hidden" lang="en"><p>We have open source’d the <a href="https://twitter.com/artsy">@Artsy</a> iOS app.&#10;&#10;<a href="https://t.co/c1SWtHmUgy">https://t.co/c1SWtHmUgy</a>&#10;&#10;🎉</p>&mdash; Ørta (@orta) <a href="https://twitter.com/orta/status/558395611754819586">January 22, 2015</a></blockquote>
+</center>
+
+
+Let's go over the process we went through to get to that point.
+
+<!-- more -->
+
+Credit where credit is due, when we were [working on Eidolon](/blog/2014/11/13/eidolon-retrospective/), our CTO [dB](http://code.dblock.org/) just casually tossed the idea that, really, Eigen should be open source too. This totally threw me for a loop, we were only just getting to a point where we could build an app from scratch in the open. Trying to get a project that had existed for years and had it's own momentum converted would take a lot of thinking about.
+
+We devoted time at the end of 2014 to understand what the constraints were for getting the app opened. From a purely functional perspective we would have to start with a [drastic decision](https://github.com/artsy/mobile/issues/11) around the repo itself. 
+
+### The Repo
+
+We opted to go for a total repo switch, removing all history. There were a lot of places where keys could have been hiding within the app. To get this done we migrated the existing `eigen` to `eigen-private` on github, and did a fresh `git init`. People who have joined Artsy Mobile post-OSS have never even cloned the `eigen-private` repo. So I feel good about this call.
+
+We used the last few moments of the private repo to remove all of the default Apple copyright notices. We didn't feel they added anything on top of the git history, and make it feel like the founders of a project are more important than anyone working on improvements.
+
+It wasn't all smooth sailing with respect to the repo switch however. As the switch happened the WatchKit came out, and we had devoted quite a lot of time to building an app in it. Given that you [can't predict Apple](http://www.elischiff.com/blog/2015/3/24/fear-of-apple)'s reactions, we opted to work on a branch from our private repo. [For months](https://github.com/artsy/eigen/pull/302). In the end it was easier to have the two folders next to each other, then copy & paste over all the files and to set all the settings in the Xcproject again.
```

Comment:
This could be clarified a bit. If I recall, there were actually problems easily removing a WatchKit extension from an Xcode project – extensions that couldn't be shipped to the store, yet.